hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenFeeScheduleUpdateHandler.java

```diff
@@ -37,15 +42,41 @@ public class TokenFeeScheduleUpdateHandler implements TransactionHandler {
      * <p>Please note: the method signature is just a placeholder which is most likely going to
      * change.
      *
-     * @param txBody the {@link TransactionBody} with the transaction data
+     * @param txn the {@link TransactionBody} with the transaction data
      * @param payer the {@link AccountID} of the payer
+     * @param keyLookup the {@link AccountKeyLookup} to use to resolve keys
+     * @param tokenStore the {@link ReadableTokenStore} to use to resolve token metadata
      * @return the {@link TransactionMetadata} with all information that needs to be passed to
      *     {@link #handle(TransactionMetadata)}
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     public TransactionMetadata preHandle(
-            @NonNull final TransactionBody txBody, @NonNull final AccountID payer) {
-        throw new UnsupportedOperationException("Not implemented");
+            @NonNull final TransactionBody txn,
+            @NonNull final AccountID payer,
+            @NonNull final AccountKeyLookup keyLookup,
+            @NonNull final ReadableTokenStore tokenStore) {
+        final var op = txn.getTokenFeeScheduleUpdate();
+        final var tokenId = op.getTokenId();
+        final var meta =
+                new SigTransactionMetadataBuilder(keyLookup).payerKeyFor(payer).txnBody(txn);
+        final var tokenMeta = tokenStore.getTokenMeta(tokenId);
+        if (tokenMeta.failed()) {
+            meta.status(tokenMeta.failureReason());
+        } else {
+            final var tokenMetadata = tokenMeta.metadata();
```

Comment:
I have no idea if the logic of this method is correct ;) I can only review the code syntax / architecture

Reply:
@hendrikebbers All the logic is being ported from `SigRequirements` class :)